src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -12,27 +12,38 @@ namespace OrchardCore.ContentManagement.GraphQL.Queries.Types
 {
     public class DynamicContentTypeBuilder : IContentTypeBuilder
     {
+        // If registered, then fields input should be exposed, otherwise ignored.
+        public class FieldsInputMarker
+        {
+        }
+
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly GraphQLContentOptions _contentOptions;
         protected readonly IStringLocalizer S;
+        private readonly FieldsInputMarker _fieldsInputMarker;
+
         private readonly Dictionary<string, FieldType> _dynamicPartFields;
 
         public DynamicContentTypeBuilder(IHttpContextAccessor httpContextAccessor,
             IOptions<GraphQLContentOptions> contentOptionsAccessor,
-            IStringLocalizer<DynamicContentTypeBuilder> localizer)
+            IStringLocalizer<DynamicContentTypeBuilder> localizer,
+            FieldsInputMarker fieldsInputMarker = null)
         {
             _httpContextAccessor = httpContextAccessor;
             _contentOptions = contentOptionsAccessor.Value;
             _dynamicPartFields = [];
 
             S = localizer;
+            _fieldsInputMarker = fieldsInputMarker;
         }
 
         public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefinition, ContentItemType contentItemType)
         {
             var serviceProvider = _httpContextAccessor.HttpContext.RequestServices;
             var contentFieldProviders = serviceProvider.GetServices<IContentFieldProvider>().ToList();
 
+            var whereInputType = (ContentItemWhereInput)contentQuery.Arguments?.FirstOrDefault(x => x.Name == "where")?.ResolvedType;
```

Comment:
There may be null references, check if they are null first

Reply:
As a nitpicky suggestion, all of the above code should be called after the `_contentOptions.ShouldHide(contentTypeDefinition)` call, which isn't rigorous enough, but you can reorder them.😊

Reply:
I injected content field providers in the constructor, you can review the last change